### PR TITLE
Adding detector for Nvidia NGC

### PR DESCRIPTION
**Pull request by JosephTLucas**
Reference: https://github.com/trufflesecurity/trufflehog/issues/790

### CODE REVIEW

File: pkg/detectors/ngc/ngc.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+	matches := keyPat1.FindAllStringSubmatch(dataStr, -1)
+
+	for _, match := range matches {
+		if len(match) != 2 {
+			continue
+		}
+		resMatch := strings.TrimSpace(match[1])
+		decode, _ := base64.StdEncoding.DecodeString(resMatch)
+
+		contains_key := keyPat2.MatchString(string(decode))
```
Comment (ahrav) [inline pkg/detectors/ngc/ngc.go:47]: nit: Looks like a snake case var name slipped in here.

**Review by ahrav — APPROVED** [review]
LGTM, thanks a lot for the contribution 😄

File: pkg/detectors/ngc/ngc.go (hunk 1 above)
Comment (JosephTLucas) [reply pkg/detectors/ngc/ngc.go]: Ha, spot the python-user! This is my first time writing golang.

**Comment by JosephTLucas** [comment]
Hey @ahrav , I'm not bumping this in priority, but I want to understand the path from here. Is it just as simple as you merge whenever you're ready? Or is there more involved with transitioning the test secrets to a trufflehog-owned secrets manager + anything else?

**Comment by adawalli** [comment]
This is awesome - thank you @JosephTLucas !!!

**Comment by imsatasia** [comment]
Thank you @JosephTLucas

**Pull request merged**